src/parser.rs

```diff
@@ -244,7 +244,7 @@ fn mk_int(s: &str) -> Result<i64, Error> {
         return Ok(0);
     }
 
-    str::parse::<i64>(&s).map_err(|e| {
+    str::parse::<i64>(s).map_err(|e| {
         // This is a janky workaround until ParseIntError.kind() is stable
```

Comment:
Can you tell me a bit more about why you chose to update these functions so they own their arts versus taking a reference?

Reply:
Referencing would be better. My bad. But instead of a taking in a &String taking in &str would be better